plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx

```diff
@@ -0,0 +1,237 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import {
+  ListItem,
+  ListItemIcon,
+  ListItemSecondaryAction,
+  Tooltip,
+  ListItemText,
+  makeStyles,
+  IconButton,
+  Typography,
+} from '@material-ui/core';
+import DoneIcon from '@material-ui/icons/Done';
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+import {
+  StatusError,
+  StatusWarning,
+  StatusOK,
+  useApi,
+  alertApiRef,
+} from '@backstage/core';
+import { formatDistanceToNowStrict } from 'date-fns';
+import { Incident, IncidentPhase } from '../types';
+import OpenInBrowserIcon from '@material-ui/icons/OpenInBrowser';
+import { splunkOnCallApiRef } from '../../api/client';
+import { useAsyncFn } from 'react-use';
+import { PatchIncidentRequest } from '../../api/types';
+
+const useStyles = makeStyles({
+  denseListIcon: {
+    marginRight: 0,
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  listItemPrimary: {
+    fontWeight: 'bold',
+  },
+  listItemIcon: {
+    minWidth: '1em',
+  },
+  secondaryAction: {
+    paddingRight: 48,
+  },
+});
+
+type Props = {
+  incident: Incident;
+  onIncidentAction: () => void;
+};
+
+const IncidentPhaseStatus = ({
+  currentPhase,
+}: {
+  currentPhase: IncidentPhase;
+}) => {
+  switch (currentPhase) {
+    case 'UNACKED':
+      return <StatusError />;
+    case 'ACKED':
+      return <StatusWarning />;
+    default:
+      return <StatusOK />;
+  }
+};
+
+const incidentPhaseTooltip = (currentPhase: IncidentPhase) => {
+  switch (currentPhase) {
+    case 'UNACKED':
+      return 'Triggered';
+    case 'ACKED':
+      return 'Acknowledged';
+    default:
+      return 'Resolved';
+  }
+};
+
+const IncidentAction = ({
+  currentPhase,
+  incidentNames,
+  resolveAction,
+  aknowledgeAction,
+}: {
+  currentPhase: string;
+  incidentNames: string[];
+  resolveAction: (args: PatchIncidentRequest) => void;
+  aknowledgeAction: (args: PatchIncidentRequest) => void;
+}) => {
+  switch (currentPhase) {
+    case 'UNACKED':
+      return (
+        <Tooltip title="Aknowledge" placement="top">
+          <IconButton onClick={() => aknowledgeAction({ incidentNames })}>
+            <DoneIcon />
+          </IconButton>
+        </Tooltip>
+      );
+    case 'ACKED':
+      return (
+        <Tooltip title="Resolve" placement="top">
+          <IconButton onClick={() => resolveAction({ incidentNames })}>
+            <DoneAllIcon />
+          </IconButton>
+        </Tooltip>
+      );
+    default:
+      return <></>;
+  }
+};
+
+export const IncidentListItem = ({ incident, onIncidentAction }: Props) => {
+  const classes = useStyles();
+  const createdAt = formatDistanceToNowStrict(new Date(incident.startTime!));
```

Comment:
Awareness if it applies here
https://backstage.io/docs/architecture-decisions/adrs-adr010